src/neurostatslib/glm.py

```diff
@@ -1,431 +1,598 @@
+            Expected shape: (n_timesteps, n_neurons, n_basis_input).
+        init_y :
+            Initial observation (spike counts for PoissonGLM) matrix that kickstarts the simulation.
+            Expected shape: (window_size, n_neurons).
+        coupling_basis_matrix :
+            Basis matrix for coupling, representing between-neuron couplings
+            and auto-correlations. Expected shape: (window_size, n_basis_coupling).
 
         Returns
         -------
-        simulated_spikes : (n_neurons, n_timesteps)
-            The simulated spikes.
+        simulated_activity :
+            Simulated activity (spike counts for PoissonGLMs) for each neuron over time.
+            Shape: (n_neurons, n_timesteps).
+        firing_rates :
+            Simulated rates for each neuron over time.
+            Shape: (n_neurons, n_timesteps).
 
         Raises
         ------
         NotFittedError
-            If ``fit`` has not been called first with this instance.
+            If the model hasn't been fitted prior to calling this method.
         ValueError
-            If attempting to simulate a different number of neurons than were
-            present during fitting (i.e., if ``init_spikes.shape[0] !=
-            self.baseline_log_fr_.shape[0]``) or if ``init_spikes`` has the
-            wrong number of time steps (i.e., if ``init_spikes.shape[1] !=
-            self.spike_basis_matrix.shape[1]``)
+            - If the instance has not been previously fitted.
+            - If there's an inconsistency between the number of neurons in model parameters.
+            - If the number of neurons in input arguments doesn't match with model parameters.
+
 
         See Also
         --------
-        predict
-            Predict firing rates based on fit parameters, for checking against existing data.
+        [predict](./#neurostatslib.glm.GLM.predict) :
+        Method to predict rates based on the model's parameters.
+
+        Notes
+        -----
+        The model coefficients (`self.basis_coeff_`) are structured such that the first set of coefficients
+        (of size `n_basis_coupling * n_neurons`) are interpreted as the weights for the recurrent couplings.
+        The remaining coefficients correspond to the weights for the feed-forward input.
 
+
+        The sum of `n_basis_input` and `n_basis_coupling * n_neurons` should equal `self.basis_coeff_.shape[1]`
+        to ensure consistency in the model's input feature dimensionality.
         """
-        try:
-            Ws = self.spike_basis_coeff_
-        except AttributeError:
-            raise NotFittedError(
-                "This GLM instance is not fitted yet. Call 'fit' with appropriate arguments."
-            )
-        bs = self.baseline_log_fr_
+        # check if the model is fit
+        self._check_is_fit()
+
+        # convert to jnp.ndarray
+        (coupling_basis_matrix,) = self._convert_to_jnp_ndarray(coupling_basis_matrix)
```

Comment:
any other validation of this matrix needs to be done? e.g., do we only allow 0s and 1s or arbitrary values?

Reply:
the shape of `coupling_basis_matrix` is checked, the number of the number of parameters of the feedforward input must be `params[0].shape[1] - n_neuron * coupling_basis_matrix.shape[1]`, which is the total number of weights for a neuron - the number of weights for the recurrent connectivity, otherwise it will result in an error.

The coupling matrix should be generally of float, and if not, it may create issues. I will enforce the conversion to float